src/deploy/FlapperInit.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.8.0;
+
+import { DssInstance } from "dss-test/MCD.sol";
+import { FlapperInstance } from "./FlapperInstance.sol";
+
+interface FlapperUniV2Like {
+    function vat() external view returns (address);
+    function daiJoin() external view returns (address);
+    function spotter() external view returns (address);
+    function rely(address) external;
+    function file(bytes32, uint256) external;
+    function file(bytes32, address) external;
+}
+
+interface FlapperMomLike {
+    function setAuthority(address) external;
+}
+
+interface OracleWrapperLike {
+    function pip() external view returns (address);
+}
+
+interface PipLike {
+    function kiss(address) external;
+}
+
+struct FlapperUniV2Config {
+    uint256 hop;
+    uint256 want;
+    address pip;
+    uint256 bump;
+}
+
+library FlapperInit {
+    uint256 constant WAD = 10 ** 18;
+
+    function initFlapperUniV2(
+        DssInstance        memory dss,
+        FlapperInstance    memory flapperInstance,
+        FlapperUniV2Config memory cfg
+    ) internal {
+        FlapperUniV2Like flapper = FlapperUniV2Like(flapperInstance.flapper);
+        FlapperMomLike   mom     = FlapperMomLike(flapperInstance.mom);
+
+        // Sanity checks
+        require(flapper.vat()     == address(dss.vat),     "Flapper vat mismatch");
+        require(flapper.daiJoin() == address(dss.daiJoin), "Flapper daiJoin mismatch");
+        require(flapper.spotter() == address(dss.spotter), "Flapper spotter mismatch");
+
+        require(cfg.hop >= 5 minutes, "hop too low");
+        require(cfg.want >= WAD * 90 / 100, "want too low");
+
+        flapper.file("hop",  cfg.hop);
+        flapper.file("want", cfg.want);
+        flapper.file("pip",  cfg.pip);
+        flapper.rely(address(dss.vow));
+        flapper.rely(address(mom));
+
+        dss.vow.file("flapper", address(flapper));
+        dss.vow.file("bump",    cfg.bump);
```

Comment:
We could probably add the hump as well. It is a value exclusive of the `flap()` execution and in fact we will need to modify it when the flapper is on production.

Reply:
Nvm, that happens due to commenting line by line without the full prev read.

Reply:
Well I might still feel that is better to keep in the same init function and not separated.

Reply:
what about changing the flapper? I can imagine that happening.

Reply:
Actually we might be fine just having an off-boarding function. So when we want to do a replacement we can off-board the actual and then do the onboarding of the new one. As the new Flapper might have other settings seems a replacement function might not be possible to generalize.

Reply:
I meant that I can imagine having a new flapper, with a modified `initFlapper` function.
In the spell where that new flapper is changed `vow.hump` should not be touched. That's why I took it out.
Let me know if you feel strongly about that.

Reply:
Hmm `bump` could also not change and we still have it in the general `init` function. It is true that `hump` might be a bit less probably that changes, but yeah just not sure, I guess the other option is to include them and spell crafter could pass the actual values.

Reply:
Ok, merged it to `initFlapperUniV2`.
I guess it's ok as the spell tests also check vow.hump value, so unlikely it will be changed to an unexpected value.